src/toil/leader.py

```diff
@@ -260,7 +260,12 @@ def innerLoop(self):
                             continue
 
                         # If the job is a checkpoint and has remaining retries then reissue it.
-                        elif jobGraph.checkpoint is not None and jobGraph.remainingRetryCount > 0:
+                        # The logic behind using > 1 rather than > 0 here: Since this job has
+                        # been tried once (without decreasing its retry count as the job
+                        # itself was successful), and its subtree failed, it shouldn't be retried
+                        # unless it has more than 1 try.
+                        elif jobGraph.checkpoint is not None and jobGraph.remainingRetryCount > 1:
```

Comment:
Nah, it's a *try* count, despite the name. So this code should be correct.

We could rename to remainingTryCount. But I think that should be a new PR. Plus it breaks jobStore resumability, and if we're going to do that, then we might as well rename many things at the same time to make it less painful.